x-pack/plugins/data_enhanced/server/search/eql_search_strategy.ts

```diff
@@ -27,56 +31,44 @@ export const eqlSearchStrategyProvider = (
         id,
       });
     },
-    search: (request, options, context) =>
-      from(
-        new Promise<EqlSearchStrategyResponse>(async (resolve, reject) => {
-          logger.debug(`_eql/search ${JSON.stringify(request.params) || request.id}`);
-          let promise: TransportRequestPromise<ApiResponse>;
 
-          try {
-            const eqlClient = context.core.elasticsearch.client.asCurrentUser.eql;
-            const uiSettingsClient = await context.core.uiSettings.client;
-            const asyncOptions = getAsyncOptions();
-            const searchOptions = toSnakeCase({ ...request.options });
+    search: (request, options, context) => {
+      logger.debug(`_eql/search ${JSON.stringify(request.params) || request.id}`);
 
-            if (request.id) {
-              promise = eqlClient.get(
-                {
-                  id: request.id,
-                  ...toSnakeCase(asyncOptions),
-                },
-                searchOptions
-              );
-            } else {
-              const { ignoreThrottled, ignoreUnavailable } = await getDefaultSearchParams(
-                uiSettingsClient
-              );
-              const searchParams = toSnakeCase({
-                ignoreThrottled,
-                ignoreUnavailable,
-                ...asyncOptions,
-                ...request.params,
-              });
+      const { esSearch } = search;
+      const asyncOptions = getAsyncOptions();
 
-              promise = eqlClient.search(
-                searchParams as EqlSearchStrategyRequest['params'],
-                searchOptions
+      return config$.pipe(
+        mergeMap(
+          () =>
+            new Promise<DoSearchFnArgs>(async (resolve) => {
+              const { ignoreThrottled, ignoreUnavailable } = await getDefaultSearchParams(
+                context.core.uiSettings.client
               );
-            }
-
-            const rawResponse = await shimAbortSignal(promise, options?.abortSignal);
-            const { id, is_partial: isPartial, is_running: isRunning } = rawResponse.body;
 
-            resolve({
-              id,
-              isPartial,
-              isRunning,
-              rawResponse,
-            });
-          } catch (e) {
-            reject(e);
-          }
-        })
-      ),
+              resolve({
+                params: {
+                  ignoreThrottled,
+                  ignoreUnavailable,
+                  ...asyncOptions,
+                  ...request.params,
+                },
+                options: { ...request.options },
+              });
+            })
+        ),
```

Comment:
Nit: This (and other places using the `new Promise` logic) can be simplified to something like this:

```ts
        mergeMap(async () => {
          const { ignoreThrottled, ignoreUnavailable } = await getDefaultSearchParams(
            context.core.uiSettings.client
          );
          return {
            params: {
              ignoreThrottled,
              ignoreUnavailable,
              ...asyncOptions,
              ...request.params,
            },
            options: { ...request.options },
          };
        }),
```